controlplane/kubeadm/internal/controllers/helpers_test.go

```diff
@@ -374,25 +374,26 @@ func TestCloneConfigsAndGenerateMachine(t *testing.T) {
 	g.Expect(env.GetAPIReader().List(ctx, machineList, client.InNamespace(cluster.Namespace))).To(Succeed())
 	g.Expect(machineList.Items).To(HaveLen(1))
 
-	for _, m := range machineList.Items {
-		g.Expect(m.Namespace).To(Equal(cluster.Namespace))
-		g.Expect(m.Name).NotTo(BeEmpty())
-		g.Expect(m.Name).To(HavePrefix(kcp.Name))
+	for i := range machineList.Items {
+		mc := machineList.Items[i]
+		g.Expect(mc.Namespace).To(Equal(cluster.Namespace))
+		g.Expect(mc.Name).NotTo(BeEmpty())
+		g.Expect(mc.Name).To(HavePrefix(kcp.Name))
 
-		infraObj, err := external.Get(ctx, r.Client, &m.Spec.InfrastructureRef, m.Spec.InfrastructureRef.Namespace)
+		infraObj, err := external.Get(ctx, r.Client, &mc.Spec.InfrastructureRef, mc.Spec.InfrastructureRef.Namespace)
 		g.Expect(err).ToNot(HaveOccurred())
 		g.Expect(infraObj.GetAnnotations()).To(HaveKeyWithValue(clusterv1.TemplateClonedFromNameAnnotation, genericInfrastructureMachineTemplate.GetName()))
 		g.Expect(infraObj.GetAnnotations()).To(HaveKeyWithValue(clusterv1.TemplateClonedFromGroupKindAnnotation, genericInfrastructureMachineTemplate.GroupVersionKind().GroupKind().String()))
 
-		g.Expect(m.Spec.InfrastructureRef.Namespace).To(Equal(cluster.Namespace))
-		g.Expect(m.Spec.InfrastructureRef.Name).To(HavePrefix(genericInfrastructureMachineTemplate.GetName()))
-		g.Expect(m.Spec.InfrastructureRef.APIVersion).To(Equal(genericInfrastructureMachineTemplate.GetAPIVersion()))
-		g.Expect(m.Spec.InfrastructureRef.Kind).To(Equal("GenericInfrastructureMachine"))
+		g.Expect(mc.Spec.InfrastructureRef.Namespace).To(Equal(cluster.Namespace))
```

Comment:
I think this should be fine if the same variable name - `m` - is kept. Can we revert this part of the change?

Reply:
done 👍